compiler/next/include/chpl/queries/ErrorMessage.h

```diff
@@ -54,6 +57,21 @@ class ErrorMessage final {
 #endif
   ;
 
+  void addDetail(ErrorMessage err);
+
+  Location location() const { return location_; }
+
+  UniqueString path() const { return location_.path(); }
+  int firstLine() const { return location_.firstLine(); }
+  int firstColumn() const { return location_.firstColumn(); }
+  int lastLine() const { return location_.lastLine(); }
+  int lastColumn() const { return location_.lastColumn(); }
+  int line() const { return location_.line(); }
```

Comment:
Not a big deal, but if it was me, I'd leave these out from the ErrorMessage and let the user do

```c++
myError->location()->firstLine()
```

which is more typing but reads more naturally to me than

```c++
myError->firstLine()
```

Reply:
I think I'll leave this one alone for now but we might develop a better idea what to do with them as the code evolves.